cli/src/commands/developer/execute.rs

```diff
@@ -33,56 +37,69 @@ use snarkvm::{
 use colored::Colorize;
 use std::{path::PathBuf, str::FromStr};
+use tracing::debug;
+use ureq::http::Uri;
 use zeroize::Zeroize;
 
 /// Executes an Aleo program function.
 #[derive(Debug, Parser)]
+#[command(
+    group(clap::ArgGroup::new("mode").required(true).multiple(false)),
+    group(clap::ArgGroup::new("key").required(true).multiple(false))
+)]
 pub struct Execute {
     /// The program identifier.
+    #[clap(value_parser=NonEmptyStringValueParser::default())]
     program_id: String,
     /// The function name.
+    #[clap(value_parser=NonEmptyStringValueParser::default())]
     function: String,
     /// The function inputs.
     inputs: Vec<String>,
-    /// Specify the network to create an execution for.
-    #[clap(default_value = "0", long = "network")]
-    pub network: u16,
     /// The private key used to generate the execution.
-    #[clap(short, long)]
+    #[clap(short = 'p', long, group = "key", value_parser=NonEmptyStringValueParser::default())]
     private_key: Option<String>,
     /// Specify the path to a file containing the account private key of the node
-    #[clap(long)]
+    #[clap(long, group = "key", value_parser=NonEmptyStringValueParser::default())]
     private_key_file: Option<String>,
-    /// The endpoint to query node state from.
-    #[clap(short, long)]
-    query: String,
+    /// Use a developer validator key to generate the execution
+    #[clap(long, group = "key")]
+    dev_key: Option<u16>,
+    /// The endpoint to query node state from and broadcast to (if set to broadcast).
+    #[clap(short, long, default_value=DEFAULT_ENDPOINT)]
+    endpoint: Uri,
     /// The priority fee in microcredits.
-    #[clap(long)]
-    priority_fee: Option<u64>,
+    #[clap(long, default_value_t = 0)]
+    priority_fee: u64,
     /// The record to spend the fee from.
     #[clap(short, long)]
     record: Option<String>,
     /// The endpoint used to broadcast the generated transaction.
-    #[clap(short, long, conflicts_with = "dry_run")]
-    broadcast: Option<String>,
+    #[clap(short, long, group = "mode")]
+    broadcast: bool,
     /// Performs a dry-run of transaction generation.
-    #[clap(short, long, conflicts_with = "broadcast")]
+    #[clap(short, long, group = "mode")]
     dry_run: bool,
     /// Store generated deployment transaction to a local file.
-    #[clap(long)]
+    #[clap(long, group = "mode")]
     store: Option<String>,
     /// If --store is specified, the format in which the transaction should be stored : string or
     /// bytes, by default : bytes.
-    #[clap(long, value_enum, default_value_t = StoreFormat::Bytes)]
+    #[clap(long, value_enum, default_value_t = StoreFormat::Bytes, requires="store")]
     store_format: StoreFormat,
-    /// Specify the path to a directory containing the ledger. Overrides the default path (also for
-    /// dev).
+    /// Wait for the transaction to be accepted by the network. Requires --broadcast.
```

Comment:
Why did you add this? Our existing shell scripts worked just fine. `snarkos developer` is just meant for very lightweight **internal** testing, Leo already exposes a much better waiting API for developers.

Reply:
I found it really useful to test https://github.com/ProvableHQ/snarkOS/pull/3638 and also reproduce the other bug fixes by this PR. Otherwise, I would have to issue the transaction and then query multiple times to see if it went through, or write a custom script.